tests/test_utilities.py

```diff
@@ -1710,6 +1711,15 @@ def test_get_active_course_runs(self, course, users_all_enrolled_courses, expect
         """
         assert utils.get_active_course_runs(course, users_all_enrolled_courses) == expected_course_run
 
+    @mock.patch('enterprise.utils.Registry')
+    def test_get_idp_choices(self, mock_registry):
+        Provider = namedtuple('SAMLProvider', 'provider_id, name, disable_for_enterprise_sso')
+        enabled_provider = Provider('12345', 'google', False)
+        disabled_provider = Provider('333333', 'facebook', True)
+
+        mock_registry.enabled.return_value = [enabled_provider, disabled_provider]
```

Comment:
is `enabled` a property on the Registry? I did not quite make the connectoin between the enabled and how get_idp_choices` works, but if this works in testing I guess it's just me trying to understand this better

Reply:
Enabled appears to be a method in the Registry class. I honestly could not find the actual code that creates this class. It is somewhere deep in one of the authentication django libraries I assume (I tried the social auth and related repos, but couldn't find Registry defined).
It is called in the `get_idp_choices` method as in: `for idp in Registry.enabled()`
The output of that call is in this format: ``` [<SAMLProviderConfig: SAMLProviderConfig object (149)>, <SAMLProviderConfig: SAMLProviderConfig object (152)>, <SAMLProviderConfig: SAMLProviderConfig object (123)>]```

Reply:
ah I see, so basically enabled() return value is what we are mocking which seems like the correct abstraction anyway, thanks for explaining

Reply:
oh yea just saw that `for idp in Registry.enabled() if not`